draft-ietf-taps-arch.md

```diff
@@ -126,32 +126,37 @@ they appear in all capitals, as shown here.
 The traditional model of using sockets for networking can be represented as follows:
 
 - Applications create connections and transfer data using the socket API.
-- The socket API provides the interface to the implementations of TCP and UDP (typically implemented in the system's kernel).
-- TCP and UDP in the kernel send and receive data over the available network layer interfaces.
+- The socket API provides the interface to the implementations of TCP and UDP
+  (typically implemented in the system's kernel).
+- TCP and UDP in the kernel send and receive data over the available network
+  layer interfaces.
+- Sockets are bound directly to transport-layer and network-layer addresses,
+  obtained via a separate resolution step, usually performed by a
+  system-provided stub resolver.
 
 ~~~~~~~~~~
 
 +-----------------------------------------------------+
 |                    Application                      |
 +-----------------------------------------------------+
-           |                             |
-+---------------------+       +-----------------------+
-|  Socket Stream API  |       |  Socket Datagram API  |
-+---------------------+       +-----------------------+
-           |                             |
-+-----------------------------------------------------+
-|         TCP                           UDP           |
-|           Kernel Protocol Implementation            |
-+-----------------------------------------------------+
-                          |
+        |                 |                  |
+  +------------+     +------------+    +--------------+
+  |    stub    |     | Stream API |    | Datagram API |
+  |  resolver  |     +------------+    +--------------+
+  +------------+          |                  |
+        |           +---------------------------------+
+        |           |     TCP                UDP      |
+        |           |  Kernel Protocol Implementation |
+        |           +---------------------------------+
+        |                           |
```

Comment:
This looks strange to me. It looks like the stub resolver bypasses the kernel protocol implementations. It should also rather run on top.

Reply:
Yep, tweaked this.

Reply:
still looks strange… is it not rather that way:
```suggestion
        |                  |                  |
  +------------+     +------------+    +--------------+
  |    stub    |     | Stream API |    | Datagram API |
  |  resolver  |     +------------+    +--------------+
  +------------+           |                  |
  |    DNS.    |           |                  |
  +------------+--------------------------------------+
  |                       TCP                UDP      |
  |              Kernel Networking Stack              |
  +---------------------------------------------------+```

Reply:
Agreed with @philsbln that the DNS resolver doesn't belong in the kernel.

How about just this:

```
        |                 |                  |
   +------------+     +------------+    +--------------+
   |    stub    |     | Stream API |    | Datagram API |
   |  resolver  |     +------------+    +--------------+
   +------------+         |                  |
                     +---------------------------------+
                     |    TCP                UDP       |
                     |    Kernel Networking Stack      |
                     +---------------------------------+
                                      |
```

Reply:
works for me either way